sky/utils/controller_utils.py

```diff
@@ -122,6 +123,42 @@ def from_name(cls, name: Optional[str]) -> Optional['Controllers']:
         return None
 
 
+# Install cli dependencies. Not using SkyPilot wheels because the wheel
+# can be cleaned up by another process.
+# TODO(zhwu): Keep the dependencies align with the ones in setup.py
+def _get_cloud_dependencies_installation_commands(
+        controller_type: str) -> List[str]:
+    commands = [
+        # aws
+        'pip list | grep boto3 > /dev/null 2>&1 || '
+        'pip install "urllib3<2" awscli>=1.27.10 botocore>=1.29.10 '
+        'boto3>=1.26.1 > /dev/null 2>&1',
+        # gcp
+        'pip list | grep google-api-python-client > /dev/null 2>&1 || '
+        'google-api-python-client>=2.69.0 google-cloud-storage '
+        '> /dev/null 2>&1',
+        f'{gcp.GOOGLE_SDK_INSTALLATION_COMMAND}',
+        # ibm
+        'pip list | grep ibm-cloud-sdk-core > /dev/null 2>&1 || '
+        'pip install ibm-cloud-sdk-core ibm-vpc ibm-platform-services '
+        'ibm-cos-sdk > /dev/null 2>&1',
+        # oci
+        'pip list | grep oci > /dev/null 2>&1 || '
+        'pip install oci > /dev/null 2>&1'
+        # kubernetes
+        'pip list | grep kubernetes > /dev/null 2>&1 || '
+        'pip install kubernetes > /dev/null 2>&1',
```

Comment:
K8s also does not support spot instances? Should we move these dependencies to L155?